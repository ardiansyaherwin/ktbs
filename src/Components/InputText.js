import React, {Component} from 'react'
import PropTypes from 'prop-types'

class InputText extends Component {
  constructor (props) {
    super(props)
    this.state = {
      value: props.value
    }
  }
  onChange (value) {
    this.setState({value})
    this.props.onChange(value)
  }
  render () {
    const {value} = this.state
    const {placeholder, inputType, style, readOnly, disabled, inputRef} = this.props
    const inputTypeValue = inputType === undefined ? 'text' : inputType
    
    return (
      <div>
        <input
        	ref={inputRef}
          	value={value}
          	onChange={(e) => this.onChange(e.target.value)}
          	onBlur={() => this.props.onBlur()}
         	onFocus={() => this.props.onFocus()}
          	placeholder={placeholder}
          	style={style}
          	readOnly={readOnly}
          	disabled={disabled}
          	type={inputTypeValue}
          	onKeyDown={(e) => this.props.onKeyDown !== undefined ? this.props.onKeyDown(e, inputTypeValue) : ''}
        />
      </div>
    )
  }
}

InputText.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  placeholder: PropTypes.string,
  type: PropTypes.oneOf(['default', 'shadow', 'number']),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onKeyDown: PropTypes.func,
  style: PropTypes.object,
  inputType: PropTypes.string,
  readOnly: PropTypes.bool,
  inputRef: PropTypes.object,
}

InputText.defaultProps = {
  onChange: () => {},
  onBlur: () => {},
  onFocus: () => {},
  onKeyDown: () => {},
  value: '',
  type: 'default',
  readOnly: false,
}

export default InputText
