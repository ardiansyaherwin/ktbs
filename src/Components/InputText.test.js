import React from 'react'
import InputText from './InputText'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

test('Input should accept default or shadow as type', () => {
  const defaultInput = renderer.create(<InputText />).toJSON()

  expect(defaultInput).toMatchSnapshot()
})

test('Input value should change when user type', () => {
  const wrapper = shallow(<InputText />)
  wrapper.find('input').simulate('change', {target:{value: 'Test 1'}})
  expect(wrapper.state('value')).toEqual('Test 1')
})