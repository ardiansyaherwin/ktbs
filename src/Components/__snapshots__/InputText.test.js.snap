// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should accept default or shadow as type 1`] = `
<div>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    readOnly={false}
    type="text"
    value=""
  />
</div>
`;
